### ISSUE TITLE
Support for java Duration 

### ISSUE THREAD

**Issue by B-hamza** [joelittlejohn/jsonschema2pojo#1409]
Json schema Spec support format duration as this issue mention https://github.com/json-schema-org/json-schema-spec/issues/565
Is it possible to have java time Duration generated for this format, as it the case for uuid format ?

**Comment by joelittlejohn** [comment]
Should be an easy PR to add. Check out the FormatRule and FormatIT.

**Comment by unkish** [comment]
It should be noted though that neither `java.time.Duration` nor `java.time.Period` are fully ISO 8601 compliant.
Links in issue mentions durations cited by https://www.rfc-editor.org/rfc/rfc3339.html#page-12

> Durations:
>
>   dur-second        = 1*DIGIT "S"
   dur-minute        = 1*DIGIT "M" [dur-second]
   dur-hour          = 1*DIGIT "H" [dur-minute]
   dur-time          = "T" (dur-hour / dur-minute / dur-second)
   dur-day           = 1*DIGIT "D"
   dur-week          = 1*DIGIT "W"
   dur-month         = 1*DIGIT "M" [dur-day]
   dur-year          = 1*DIGIT "Y" [dur-month]
   dur-date          = (dur-day / dur-month / dur-year) [dur-time]
>
>   duration          = "P" (dur-date / dur-time / dur-week)

For example it is not possible to construct `java.time.Duration` of values:

- P1W
- P1Y
- ...

**Comment by eirnym** [comment]
@unkish agree. But for the most use cases duration type is good enough. Like default regex implementation as I wrote below. 

There's way more to add about compliance of Java types to JSON schema standards. E.g. one of such type types is regex: Java's version still support regexes like `{3}` (this is the full regex) and doesn't support some features of ECMA's. In this case to support it, I had to add jruby's implementation. 

If parser-specific formatter exists in a parser library, it's better to add an annotation for it. Otherwise, I'd add parser-specific things to a parser or a separate library.

**Comment by unkish** [comment]
I'm sorry, maybe I misunderstand what You are trying to hint at, the following won't work in Jackson json:
```
final ObjectMapper om = new ObjectMapper().registerModule(new JavaTimeModule());
final String p1y2mJson = om.writeValueAsString(Period.parse("P1Y2M"));
final Duration pAsDuration = om.readValue(p1y2mJson, Duration.class);
```

**Comment by joelittlejohn** [comment]
Out if interest, does this work:

```java
om.readValue("P1Y2M", TemporalAmount.class);
```

?

**Comment by unkish** [comment]
Unfortunately not:

> Cannot construct instance of `java.time.temporal.TemporalAmount` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information

I can't imagine what it would return in case something more complex such as `P1Y2MT4H5M` - `java.time.Duration` ?
